versioned_docs/version-V3/guides/swaps/single-swaps.md

```diff
@@ -4,276 +4,331 @@ title: Single Swaps
-    /// @param amountIn The exact amount of DAI that will be swapped for WETH9.
-    /// @return amountOut The amount of WETH9 received.
-    function swapExactInputSingle(uint256 amountIn) external returns (uint256 amountOut) {
-        // msg.sender must approve this contract
+function swapWETHForDAI(uint amountIn) external returns (uint256 amountOut) {
+    // Transfer the specified amount of WETH9 to this contract.
+    TransferHelper.safeTransferFrom(WETH9, msg.sender, address(this), amountIn);
+    
+    // Approve the router to spend WETH9.
+    TransferHelper.safeApprove(WETH9, address(swapRouter), amountIn);
+}
+```
+
+The first step calls the periphery helper function `safeTransferFrom`, which transfers the desired amount of WETH from the caller’s wallet into the contract. Keep in mind that since we’re transferring on behalf of the user, that user will have to sign an approval before calling this method. This is a critical concept to understand, and we’ll go through *how* to do that in the next section when we create a test client. 
 
-        // Transfer the specified amount of DAI to this contract.
-        TransferHelper.safeTransferFrom(DAI, msg.sender, address(this), amountIn);
+The next line calls `safeApprove` to allow the `swapRouter` to spend the specified amount of WETH. This will give SwapRouter permission to actually execute the swap for us. 
 
-        // Approve the router to spend DAI.
-        TransferHelper.safeApprove(DAI, address(swapRouter), amountIn);
+We’ll call the SwapRouter’s [exactInputSingle](https://docs.uniswap.org/protocol/reference/periphery/interfaces/ISwapRouter#exactinputsingle) method, which will run a swap of an “exact amount” of an input token for the maximum amount of an output token. 
 
+Add the following code to the `swapWETHForDAI` function to execute the swap. It may seem complicated, but we’ll step through it.  
+
+```solidity
+function swapWETHForDAI(uint amountIn) external returns (uint256 amountOut) {
+	// ...
+	ISwapRouter.ExactInputSingleParams memory params =
+      ISwapRouter.ExactInputSingleParams({
+          tokenIn: WETH9,
+          tokenOut: DAI,
+          fee: feeTier,
+          recipient: msg.sender,
+          deadline: block.timestamp,
+          amountIn: amountIn,
+          amountOutMinimum: 0,
+          sqrtPriceLimitX96: 0
+      });
+  // The call to `exactInputSingle` executes the swap.
+  amountOut = swapRouter.exactInputSingle(params);
+  return amountOut;
+}
 ```
 
-### Swap Input Parameters
+The V3-Periphery contracts provide us a [handy struct](https://github.com/Uniswap/v3-periphery/blob/9ca9575d09b0b8d985cc4d9a0f689f7a4470ecb7/contracts/interfaces/ISwapRouter.sol#L10), `ExactInputSingleParams` for the Swap method. All but the last two elements of this object should be pretty self explanatory, we’re just mapping variables that we already set to the parameter object. 
```

Comment:
need to state that this structure is for the original v3 swap router 0xE592427A0AEce92De3Edee1F18E0157C05861564
new swap router has different structure